src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpPreferAsSpanOverSubstring.Fixer.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.NetCore.Analyzers.Runtime;
+
+namespace Microsoft.NetCore.CSharp.Analyzers.Runtime
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp)]
+    public sealed class CSharpPreferAsSpanOverSubstringFixer : PreferAsSpanOverSubstringFixer
+    {
+        private protected override void ReplaceNonConditionalInvocationMethodName(SyntaxEditor editor, SyntaxNode memberInvocation, string newName)
+        {
+            var cast = (InvocationExpressionSyntax)memberInvocation;
+
+            //  We know that cast.Expression can't be a MemberBindingExpressionSyntax because the analyzer only reports 
+            //  non-conditional substring invocations, i.e. not `foo?.Substring(1)`
+            RoslynDebug.Assert(cast.Expression is MemberAccessExpressionSyntax);
```

Comment:
This assert seems redundant given the direct cast below is exactly the same - it doesn't matter if the assert fires or a cast exception is thrown.

Reply:
Fixed.